File: sort/java/SelectionSort.java
public class SelectionSort {
    public static void sort(int[] array) {
        for (int i = 0; i < array.length; i++) {
            int minor = i;

            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[minor]) {
                    minor = j;
                }
            }

            Utils.swap(array, i, minor);
        }
    }
}

File: sort/c/selectionsort.c
#include "algorithms.h"

void selectionsort(int *array, size_t length) {
  for (size_t i = 0; i < length; i ++) {
    int minor = i;

    for (size_t j = i + 1; j < length; j++) {
      if (array[j] < array[minor]) {
        minor = j;
      }
    }

    swap(&array[i], &array[minor]);
  }
}
